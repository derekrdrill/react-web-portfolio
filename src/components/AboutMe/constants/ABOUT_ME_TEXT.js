export const ABOUT_ME_TEXT = [
  {
    id: 'paragraph1',
    text: `Simple, but creative. Calm, but witty. Passionate, but measured.`,
  },
  {
    id: 'paragraph2',
    description: 'Paragraph 1',
    text: `Hi, I'm Derek. I'm just a guy from a small town in Maryland, 
        who now resides in the beautiful city of Charlotte.
        I remember falling in love with writing code, in an Intro to VB.Net class
        at Salisbury University. It all clicked for me. I was building the cheesiest
        'Movie Night' WinForms app you've ever seen, and it all 
        just started to make sense. I felt this fiery sensation
        inside as I built feature after feature, and solved problem after problem.
        The rush was addictive, and for someone who's always valued themself in working hard,
        solving problems and helping others, this was the high for me. I knew I wanted
        this to be my career.`,
  },
  {
    id: 'paragraph3',
    text: `After undergrad, I accepted a role as a level 1 programmer,
        working for a very small IT team, assigned to maintaining 
        and building onto their ERP software. This was very much an introduction
        to what it was like to be a professional developer, and was perfect
        for me to organically build a strong foundation of development principles.
        However, I knew I had to continue to sharpen my skills in my freetime.
        By taking online course after course, I continued to advance my skills
        in technologies like HTML, CSS, JavaScript, JQuery, Bootstrap and SQL,
        but also teaching myself more modern technologies and libraries, such as,
        Knockout.js, Node.js, React JS and MongoDB. I've carried those hard-working
        values, that I enstilled early on, through my career. 5 years in, and I am now
        a Front End Developer, and have accomplished things like:`,
  },
  {
    id: 'accolade1',
    text: `- Being apart of a team that entirely 
        rebuilt a legacy, desktop ERP application
        into a powerful modern web-app`,
  },
  {
    id: 'accolade2',
    text: `- Worked with clients at academic offices
        of education institutions to completely 
        automate their paper-processes by building
        workflows that comprised of powerful 
        web forms that integrated with their live ERP data`,
  },
  {
    id: 'accolade3',
    text: `- Heavily contributed towards the development of 
        a CDN library of reusable Knockout.js components
        that made the development of powerful web forms 
        (mentioned above) easier and less time consuming`,
  },
  {
    id: 'accolade4',
    text: `- Work as a React developer on a dev team 
        that is building and maintaining the IMPLAN App,
        a powerful economic analysis software tool
        `,
  },
  {
    id: 'paragraph4',
    text: `The last bullet is my current role, a Front End developer
        for the IMPLAN Group, and even after several years of hard work and continual education
        to get to this point, I continue to push the limits of my knowledge daily, in and out of the office, 
        to continue to grow as a professional front end developer`,
  },
  {
    id: 'paragraph5',
    text: `TLDR;`,
  },
  {
    id: 'paragraph6',
    text: `I love to write code because it fulfills the burning desire within me
        to work hard and solve intricate problems for others, and as an admitted nerd who 
        is not the most artistic, it's the best avenue to flexing my creativity muscles üí™üèº`,
  },
];
