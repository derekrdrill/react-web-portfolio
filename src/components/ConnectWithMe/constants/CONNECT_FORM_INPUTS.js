export const CONNECT_FORM_INPUTS = [
  {
    id: 1,
    title: 'Basic Info Inputs',
    inputs: [
      {
        id: 1,
        label: 'First Name',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
      {
        id: 2,
        label: 'Last Name',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
      {
        id: 3,
        label: 'Email',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
      {
        id: 4,
        label: 'Phone (Optional)',
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        inputPaddingBottom: 5,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
    ],
  },
  {
    id: 2,
    title: 'Message Text Area',
    inputs: [
      {
        id: 1,
        label: `Tell me somethin' good`,
        variant: 'filled',
        xs: 12,
        fullWidth: true,
        multiline: true,
        minRows: 11,
        maxRows: 11,
        inputBackgroundColor: 'gainsboro',
        inputborderRadius: 4,
        fontFamily: 'Kufam',
        labelColor: '#28282B',
      },
    ],
  },
];
