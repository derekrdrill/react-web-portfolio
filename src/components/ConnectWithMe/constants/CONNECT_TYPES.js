import React from 'react';
import { faEnvelopeSquare } from '@fortawesome/fontawesome-free-solid';
import { faGithubAlt, faLinkedin } from '@fortawesome/free-brands-svg-icons';

export const CONNECT_TYPES = [
  {
    id: 1,
    title: 'Email',
    icon: faEnvelopeSquare,
    href: 'mailto:derekrdrill@gmail.com',
    target: '',
    labelLeft: 'Email me',
  },
  {
    id: 2,
    title: 'LinkedIn',
    icon: faLinkedin,
    href: 'https://www.linkedin.com/in/derek-drill',
    target: '_blank',
    labelRight: 'Connect with my LinkedIn',
  },
  {
    id: 3,
    title: 'Github',
    icon: faGithubAlt,
    href: 'https://github.com/derekrdrill',
    target: '_blank',
    labelLeft: 'See this project on Github',
  },
];
