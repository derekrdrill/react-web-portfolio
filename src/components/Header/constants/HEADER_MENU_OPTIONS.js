import React from 'react';
import AppsIcon from '@mui/icons-material/Apps';
import ConnectWithoutContactIcon from '@mui/icons-material/ConnectWithoutContact';
import PetsIcon from '@mui/icons-material/Pets';
import FaceIcon from '@mui/icons-material/Face';

export const HEADER_MENU_OPTIONS = [
  {
    id: 1,
    menuTitle: 'About Me',
    menuIcon: <FaceIcon />,
    mainTo: 'about-me',
    secondaryTo: '/about-me',
  },
  {
    id: 2,
    menuTitle: 'Projects',
    menuIcon: <AppsIcon />,
    mainTo: 'projects',
    secondaryTo: '/projects',
  },
  {
    id: 3,
    menuTitle: 'Connect With Me',
    menuIcon: <ConnectWithoutContactIcon />,
    mainTo: 'connect-with-me',
    secondaryTo: '/connect-with-me',
  },
  {
    id: 4,
    menuTitle: 'George',
    menuIcon: <PetsIcon />,
    mainTo: 'home',
    secondaryTo: '/',
  },
];
