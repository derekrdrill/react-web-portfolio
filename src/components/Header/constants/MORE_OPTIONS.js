import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEnvelopeSquare, faFile, faCode } from '@fortawesome/fontawesome-free-solid';
import { faGithubAlt, faLinkedin } from '@fortawesome/free-brands-svg-icons';

export const MORE_OPTIONS = [
  {
    id: 1,
    title: 'Email',
    icon: <FontAwesomeIcon icon={faEnvelopeSquare} />,
    href: 'mailto:derekrdrill@gmail.com',
    target: '',
  },
  {
    id: 2,
    title: 'LinkedIn',
    icon: <FontAwesomeIcon icon={faLinkedin} />,
    href: 'https://www.linkedin.com/in/derek-drill',
    target: '_blank',
  },
  {
    id: 3,
    title: 'Github',
    icon: <FontAwesomeIcon icon={faGithubAlt} />,
    href: 'https://github.com/derekrdrill',
    target: '_blank',
    divider: true,
  },
  {
    id: 4,
    title: 'Resume',
    icon: <FontAwesomeIcon icon={faFile} />,
    target: '',
    href: '',
    modal: {
      modalTitle: 'Select one',
      modalDescription: 'Would you like to download a copy or be sent to the Google doc?',
      modalButton1: 'Download .DOCX',
      modalButton2: 'Go to Google Doc',
      modalCloseButton: 'Cancel',
    },
  },
  {
    id: 5,
    title: 'Source Code',
    icon: <FontAwesomeIcon icon={faCode} />,
    target: '',
    href: '',
  },
];
