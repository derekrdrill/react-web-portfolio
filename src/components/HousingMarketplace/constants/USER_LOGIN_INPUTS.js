export const USER_LOGIN_INPUTS = [
  {
    id: 'username',
    label: 'Username',
    variant: 'outlined',
    xs: 12,
    fullWidth: true,
  },
  {
    id: 'password',
    label: 'Password',
    variant: 'outlined',
    xs: 12,
    fullWidth: true,
    type: 'password',
  },
];
