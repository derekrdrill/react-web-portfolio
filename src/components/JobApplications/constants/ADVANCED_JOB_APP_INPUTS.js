export const ADVANCED_JOB_APP_INPUTS = [
  {
    id: 1,
    title: 'Personal Information',
    inputs: [
      {
        id: 'firstName',
        label: 'First Name',
        variant: 'outlined',
        xs: 12,
        sm: 6,
        md: 3,
      },
      {
        id: 'lastName',
        label: 'Last Name',
        variant: 'outlined',
        xs: 12,
        sm: 6,
        md: 3,
      },
      {
        id: 'email',
        label: 'Email',
        variant: 'outlined',
        xs: 12,
        sm: 6,
        md: 3,
        type: 'email',
      },
      {
        id: 'phone',
        label: 'Phone',
        variant: 'outlined',
        xs: 12,
        sm: 6,
        md: 3,
      },
    ],
  },
  {
    id: 2,
    title: 'Address Information',
    inputs: [
      {
        id: 'street',
        label: 'Street',
        variant: 'outlined',
        xs: 12,
      },
      {
        id: 'city',
        label: 'City',
        variant: 'outlined',
        xs: 12,
        md: 6,
      },
      {
        id: 'state',
        label: 'State',
        variant: 'outlined',
        xs: 6,
        md: 3,
      },
      {
        id: 'zip',
        label: 'Zip',
        variant: 'outlined',
        xs: 6,
        md: 3,
      },
    ],
  },
  {
    id: 3,
    title: 'Position Information',
    inputs: [
      {
        id: 'position',
        label: 'What position are you applying for?',
        variant: 'outlined',
        sm: 6,
        md: 6,
      },
      {
        id: 'startDate',
        label: 'When can you start?',
        variant: 'outlined',
        sm: 6,
        md: 6,
      },
      {
        id: 'fullOrPart',
        label: 'Full or part time?',
        variant: 'outlined',
        select: true,
        selectOptions: [
          {
            id: 1,
            value: 'Full time',
          },
          {
            id: 2,
            value: 'Part time',
          },
        ],
        sm: 6,
        md: 6,
      },
      {
        id: 'expectedPay',
        label: 'Expected pay rate',
        variant: 'outlined',
        sm: 6,
        md: 6,
      },
      {
        id: 'crime',
        label: 'Have you ever been convicted of a crime?',
        variant: 'outlined',
        select: true,
        selectOptions: [
          {
            id: 1,
            value: 'Yes',
          },
          {
            id: 2,
            value: 'No',
          },
        ],
        sm: 6,
        md: 6,
      },
      {
        id: 'crimeExplain',
        label: 'Please explain',
        variant: 'outlined',
        multiline: true,
        sm: 6,
        md: 6,
      },
    ],
  },
  {
    id: 4,
    title: 'Qualifications',
    instructions:
      'Please list any education or training you feel relates to the position you are applying for that would help you perform the work, such as, schools, colleges, degrees, vocational or technical programs, and military training',
    dynamicList: true,
    inputs: [
      {
        id: 'institution',
        label: 'Institution',
        variant: 'outlined',
        xs: 12,
        md: 6,
      },
      {
        id: 'fromYear',
        label: 'From Year',
        variant: 'outlined',
        xs: 6,
        md: 3,
      },
      {
        id: 'toYear',
        label: 'To Year',
        variant: 'outlined',
        xs: 6,
        md: 3,
      },
      {
        id: 'degreeReceived',
        label: 'Degree Received',
        variant: 'outlined',
        xs: 12,
        sm: 6,
      },
      {
        id: 'areaOfSpecialization',
        label: 'Area of Specialization',
        variant: 'outlined',
        xs: 12,
        sm: 6,
      },
    ],
  },
  {
    id: 5,
    title: 'Special Skills',
    inputs: [
      {
        id: 'specialSkills',
        label: 'Please list any skills you feel would help you in the position you are applying for',
        multiline: true,
        minRows: 3,
        xs: 12,
      },
    ],
  },
  {
    id: 6,
    title: 'References',
    instructions: 'Please at least 3 professional or educational references',
    dynamicList: true,
    inputs: [
      {
        id: 'referenceName',
        label: 'Name',
        xs: 12,
        sm: 6,
        md: 3,
      },
      {
        id: 'referenceEmail',
        label: 'Email',
        xs: 12,
        sm: 6,
        md: 3,
      },
      {
        id: 'referencePhone',
        label: 'Phone',
        xs: 12,
        sm: 6,
        md: 3,
      },
      {
        id: 'referenceRelationship',
        label: 'Relationship',
        xs: 12,
        sm: 6,
        md: 3,
      },
    ],
  },
  {
    id: 7,
    title: 'Supporting Documents',
    instructions: (
      <div>
        These documents could include Resume CV, Cover Letter, Letter of Recomendation, Transcript, etc
        <br />
        <i>This is not required, but information in these documents could impact our decision</i>
      </div>
    ),
    dynamicList: true,
    inputs: [
      {
        id: 'fileUpload',
        type: 'file',
        xs: 12,
      },
    ],
  },
];
