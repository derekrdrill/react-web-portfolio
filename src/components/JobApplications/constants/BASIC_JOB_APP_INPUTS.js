export const BASIC_JOB_APP_INPUTS = [
  {
    id: 'name',
    label: 'Name',
    variant: 'standard',
    xs: 12,
    sm: 6,
    md: 3,
    fullWidth: true,
  },
  {
    id: 'email',
    label: 'Email',
    variant: 'standard',
    xs: 12,
    sm: 6,
    md: 3,
    fullWidth: true,
    type: 'email',
  },
  {
    id: 'phone',
    label: 'Phone',
    variant: 'standard',
    xs: 12,
    sm: 6,
    md: 3,
    fullWidth: true,
  },
  {
    id: 'job',
    label: 'What job are you applying for?',
    variant: 'standard',
    xs: 12,
    sm: 6,
    md: 3,
    fullWidth: true,
  },
  {
    id: 'summary',
    label: 'Write a short summary about yourself',
    variant: 'standard',
    xs: 12,
    fullWidth: true,
    multiline: true,
    minRows: 3,
  },
];
