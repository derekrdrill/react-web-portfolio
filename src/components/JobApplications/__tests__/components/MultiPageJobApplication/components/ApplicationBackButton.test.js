import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import {
  ApplicationBackButton,
  StyledButton,
  getBackButtonTitle,
} from '../../../../components/MultiPageJobApplication/components/ApplicationBackButton';

import { ADVANCED_JOB_APP_INPUTS } from '../../../../constants/ADVANCED_JOB_APP_INPUTS';

configure({ adapter: new Adapter() });

const renderer = new ShallowRenderer();

describe('ApplicationBackButton tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({ darkMode: true });
    renderer.render(<ApplicationBackButton />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders StyledButton styled component correctly', () => {
    const styledButton = shallow(<StyledButton />);
    const styledButtonDark = shallow(<StyledButton darkMode />);

    expect(styledButton.props().className).toEqual('sc-bcXHqe jcTuIk');
    expect(styledButtonDark.props().className).toEqual('sc-bcXHqe wrsjF');
  });

  it('runs getBackButtonTitle', () => {
    const backButtonTests = [
      { page: 1, result: `\u00A0\u00A0BACK TO LANDING PAGE` },
      { page: 2, result: `\u00A0\u00A0BACK TO ${ADVANCED_JOB_APP_INPUTS[0].title}` },
    ];

    backButtonTests.forEach(({ page, result }) => {
      expect(getBackButtonTitle(ADVANCED_JOB_APP_INPUTS, page)).toEqual(result);
    });
  });
});
