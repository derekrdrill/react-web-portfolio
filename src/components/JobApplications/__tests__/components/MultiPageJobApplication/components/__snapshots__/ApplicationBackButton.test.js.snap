// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationBackButton tests renders correctly 1`] = `
<Styled(Component)
  color="info"
  darkMode={true}
>
  <FontAwesomeIcon
    beat={false}
    beatFade={false}
    border={false}
    bounce={false}
    className=""
    fade={false}
    fixedWidth={false}
    flip={false}
    icon={
      Object {
        "icon": Array [
          320,
          512,
          Array [],
          "f053",
          "M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z",
        ],
        "iconName": "chevron-left",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    maskId={null}
    pull={null}
    pulse={false}
    rotation={null}
    shake={false}
    size={null}
    spin={false}
    spinPulse={false}
    spinReverse={false}
    swapOpacity={false}
    symbol={false}
    title=""
    titleId={null}
    transform={null}
  />
    LANDING PAGE
</Styled(Component)>
`;
